File: Solutions/1207/1207.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer,Integer> count=new HashMap<>();
        HashSet<Integer> occurences=new HashSet<>();
        for(final int num:arr)
            count.merge(num,1,Integer::sum);
        for(final int value:count.values())
            if(!occurences.add(value))
                return false;
        return true;
    }
}

File: Solutions/1207/1207.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int,int> count;
        unordered_set<int> occurences;
        for(const int num:arr)
            count[num]++;
        for(const auto& [_,value]:count)
            if(!occurences.insert(value).second)
                return false;
        return true;

    }
};